firedrake/mg/interface.py

```diff
@@ -18,20 +18,41 @@ def prolong(coarse, fine):
     hierarchy, lvl = utils.get_level(cfs)
     if hierarchy is None:
         raise RuntimeError("Coarse function not from hierarchy")
+    # for skipping hierarchies
+    if hasattr(hierarchy, '_full_hierarchy') == 1:
+        lvl = lvl * hierarchy._skip
+        hierarchy = hierarchy._full_hierarchy
```

Comment:
Yes, but firedrake doesn't know anything about that package up to now.  Now it needs to.  You should add the ability to make skipping hierarchies directly in firedrake.